spec/api/keys/keys_spec.rb

```diff
@@ -0,0 +1,498 @@
+    end
+    it "should authenticate against the key" do
+      get("#{platform.server}/organizations/#{org['name']}/clients/#{client['name']}", Pedant::Requestor.new(client['name'], $private_key)).should look_like({:status => 200})
+    end
+  end
+
+  context "when a key is expired for a user" do
+    before(:each) do
+      system("chef-server-ctl add-user-key #{user['name']} #{$public_key_filepath} --key-name key-#{Time.now.to_i} --expiration-date 2012-12-24T21:00:00")
+    end
+    it "should fail against the expired key" do
+      get("#{platform.server}/users/#{user['name']}", Pedant::Requestor.new(user['name'], $private_key)).should look_like({:status => 401})
+    end
+    it "should succeed against other keys" do
+      get("#{platform.server}/users/#{user['name']}", Pedant::Requestor.new(user['name'], user['private_key'])).should look_like({:status => 200})
+    end
+  end
+
+  context "when a key is expired for a client" do
+    before(:each) do
+      system("chef-server-ctl add-client-key #{org['name']} #{client['name']} #{$public_key_filepath} --key-name key-#{Time.now.to_i} --expiration-date 2012-12-24T21:00:00")
+    end
+    it "should fail against the expired key" do
+      get("#{platform.server}/organizations/#{org['name']}/clients/#{client['name']}", Pedant::Requestor.new(client['name'], $private_key)).should look_like({:status => 401})
+    end
+    it "should succeed against other keys" do
+      get("#{platform.server}/organizations/#{org['name']}/clients/#{client['name']}", Pedant::Requestor.new(client['name'], client['private_key'])).should look_like({:status => 200})
+    end
+  end
+
+  context "when the default key for a user exists" do
+    it "the public_key field returned by GET /users/:user and from the keys table should be the same" do
+      user_api_public_key = JSON.parse(get("#{platform.server}/users/#{user['name']}", superuser))['public_key']
+      `chef-server-ctl list-user-keys #{user['name']}`.should include(user_api_public_key)
+    end
+  end
+
+  context "when the default key for a client exists" do
+    it "should return public_key field returned by GET /organization/:org/clients/:client and from the keys table should be the same" do
+      client_api_public_key = JSON.parse(get("#{platform.server}/organizations/#{org['name']}/clients/#{client['name']}", superuser))['public_key']
+      `chef-server-ctl list-client-keys #{org['name']} #{client['name']}`.should include(client_api_public_key)
+    end
+  end
+
+  context "when the default key is updated for a user via a PUT to /users" do
+    before(:each) do
+      original_data = JSON.parse(get("#{platform.server}/users/#{user['name']}", superuser))
+      original_data['public_key'] = $public_key
+      put("#{platform.server}/users/#{user['name']}", superuser, :payload => JSON.generate(original_data))
+    end
+    context "when the default key has already been deleted" do
+      # TODO: things break
```

Comment:
TODO cleanup - a pended test perhaps? 
Current broken state: 

The key will appear to be present in the the user record returned via users endpoint, but  won't be usable for auth 
- deleted key + update to the user: The key will appear to be updated in the user record via users endpoint, but won't be usable for auth
- 
Correct behavior: ... well, we'll need to change user to return a list of associated public keys..... but other than that you should be able to test that auth of  the default key fails?